src/com/oltpbenchmark/api/BenchmarkModule.java

```diff
@@ -393,36 +392,112 @@ public void enableForeignKeys() throws Exception {
       }
 
     // This function creates SQL procedures that the execution would need. Currently we have procedures to update the
-      // Stock table, and a procedure to get stock levels of items recently ordered.
-      public void createSqlProcedures() throws Exception {
-        try {
-          Connection conn = makeConnection();
-          Statement st = conn.createStatement();
-
-          StringBuilder argsSb = new StringBuilder();
-          StringBuilder updateStatements = new StringBuilder();
-
-          argsSb.append("wid int");
-          for (int i = 1; i <= 15; ++i) {
-            argsSb.append(String.format(", i%d int, q%d int, y%d int, r%d int", i, i, i, i));
-            updateStatements.append(String.format(
-              "UPDATE STOCK SET S_QUANTITY = q%d, S_YTD = y%d, S_ORDER_CNT = S_ORDER_CNT + 1, " +
-              "S_REMOTE_CNT = r%d WHERE S_W_ID = wid AND S_I_ID = i%d;",
-              i, i, i, i));
-            String updateStmt =
-              String.format("CREATE PROCEDURE updatestock%d (%s) AS '%s' LANGUAGE SQL;",
-                            i, argsSb.toString(), updateStatements.toString());
-
-            st.execute(String.format("DROP PROCEDURE IF EXISTS updatestock%d", i));
-            st.execute(updateStmt);
-          }
+    // Stock table, and a procedure to get stock levels of items recently ordered.
+    public void createSqlProcedures() throws Exception {
```

Comment:
maybe its time to expose a `createSqlProcedures` method on `SchemaManager` and forward these calls to a `BenchmarkModule` owned instance of a `SchemaManager` generated with the `workConf`? then each base class can implement the appropriate behavior, opaque to this class, and that logic lives with other schema specification code

Reply:
Very good point! Done.